client/driver/rkt.go

```diff
@@ -563,18 +691,64 @@ func (d *RktDriver) Start(ctx *ExecContext, task *structs.Task) (*StartResponse,
 				return nil, fmt.Errorf("port_map is not set. When you defined port in the resources, you need to configure port_map.")
 			}
 
-			hostPortStr := strconv.Itoa(port.Value)
-
 			d.logger.Printf("[DEBUG] driver.rkt: exposed port %s", containerPort)
+
 			// Add port option to rkt run arguments. rkt allows multiple port args
-			prepareArgs = append(prepareArgs, fmt.Sprintf("--port=%s:%s", containerPort, hostPortStr))
+			if img != "" {
+				hostPortStr := strconv.Itoa(port.Value)
+				prepareArgs = append(prepareArgs, fmt.Sprintf("--port=%s:%s", containerPort, hostPortStr))
+			} else {
+				containerPortACName, err := appctypes.NewACName(containerPort)
+				if err != nil {
+					return nil, fmt.Errorf("[ERR] driver.rkt: failed to set ACName for %s: %s", containerPort, err)
+				}
+
+				podManifest.Ports = append(podManifest.Ports, appctypes.ExposedPort{
+					Name:     *containerPortACName,
+					HostPort: uint(port.Value),
+				})
+			}
 		}
+	}
 
+	if img != "" {
+		// If a user has been specified for the task, pass it through to the user
+		if task.User != "" {
+			prepareArgs = append(prepareArgs, fmt.Sprintf("--user=%s", task.User))
+		}
 	}
 
-	// If a user has been specified for the task, pass it through to the user
-	if task.User != "" {
-		prepareArgs = append(prepareArgs, fmt.Sprintf("--user=%s", task.User))
+	if img == "" {
```

Comment:
I think this if statement will always be true. Or am I missing something? :)

Reply:
This can be false if the pod manifest was supplied by the user instead.

Reply:
Well right above you check `if img != "" {...}`. I find this handling of conditions a bit awkward and not so clear. IMHO.